cube/src/main/java/org/jboss/arquillian/ce/cube/CEApplicationHandler.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2016 Red Hat Inc. and/or its affiliates and other
+ * contributors as indicated by the @author tags. All rights reserved.
+ * See the copyright.txt in the distribution for a full listing of
+ * individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.arquillian.ce.cube;
+
+import org.arquillian.cube.openshift.impl.client.OpenShiftClient;
+import org.jboss.arquillian.core.api.annotation.Observes;
+
+/**
+ * CEApplicationHandler
+ * <p/>
+ * Manages "applications" declared for CE test cases.
+ * 
+ * @author Rob Cernich
+ */
+public class CEApplicationHandler {
+
+    
+    public void registerCubes(@Observes OpenShiftClient client, CECubeConfiguration config) {
+        // TODO: add cubes for configured "applications"
```

Comment:
I was toying with describing the environment (application) externally with json/yaml and properties files, instead of using annotations.  That was before I realized that containers were global to all tests and our tests each use a different environment for class.  At some point, I hope to get back to an external definition...and this would be where we'd create the cubes that would define a more complex environment, and modify the createEnvironment() code to simply send StartCube() events for the specific environment.